precompiles/crowdloan-rewards/src/lib.rs

```diff
@@ -264,7 +283,20 @@ where
 			target: "crowdloan-rewards-precompile",
 			"In update_reward_address dispatchable wrapper"
 		);
-		log::trace!(target: "crowdloan-rewards-precompile", "input is {:?}", input);
+
+		// Bound check
+		if input.len() < 32 {
+			log::trace!(target: "crowdloan-rewards-precompile",
+				"Unable to parse address. Got {} bytes, expected {}",
+				input.len(),
+				32,
+			);
+			return Err(ExitError::Other(
+				"Incorrect input length for update_reward_address".into(),
+			));
+		}
```

Comment:
`parse_account` does this check already

Reply:
Not really, since we are already passing input[0..32] to parse_account. If input has less than 32 bytes this panics

Reply:
I thought it did but when testing I got a panic and realized this

Reply:
Really, that surprises me. Here is the code

```rust
fn parse_account(input: &[u8]) -> Result<H160, ExitError> {
	const PADDING_SIZE_BYTES: usize = 12;
	const ACCOUNT_SIZE_BYTES: usize = 20;
	const TOTAL_SIZE_BYTES: usize = PADDING_SIZE_BYTES + ACCOUNT_SIZE_BYTES;

	if input.len() != TOTAL_SIZE_BYTES {
		log::trace!(target: "staking-precompile",
			"Unable to parse address. Got {} bytes, expected {}",
			input.len(),
			TOTAL_SIZE_BYTES,
		);
		return Err(ExitError::Other(
			"Incorrect input length for account parsing".into(),
		));
	}

	Ok(H160::from_slice(
		&input[PADDING_SIZE_BYTES..TOTAL_SIZE_BYTES],
	))
}
```

`TOTAL_SIZE_BYTES = 32` so if the input isn't 32, then it should return 

```
log::trace!(target: "staking-precompile",
	"Unable to parse address. Got {} bytes, expected {}",
	input.len(),
	TOTAL_SIZE_BYTES,
);
return Err(ExitError::Other(
	"Incorrect input length for account parsing".into(),
));
```

Reply:
But if you call it like this 

`let collator = parse_account(&input[..32])?;`

Then if  the length is less than 32 this panics before reaching the function

Reply:
This is resolved, I'm just keeping it open to remember it later.